.github/workflows/spellcheck.yml

```diff
@@ -0,0 +1,26 @@
+name: Run spellchecker for latex and markdown files
+on: 
+  push:
+    paths:
+      - '**.md'
+      - '**.tex'
+      - '.github/workflows/spellcheck.yml'
+      - 'scripts/spellcheck.sh'
+
+jobs:
+  spellcheck:
+    runs-on: ubuntu-latest
+    name: Spellchecker
+
+    steps:
+    - name: Download repo
+      uses: actions/checkout@v2
+
+    - name: Install spellchecker tool
```

Comment:
If this takes too long, I'd go for a Docker container or caching the installation

Reply:
It goes really quick to be honest, most of the times I don't even get to see the action being executed, but will consider it in case it changes, thanks!